### DISCUSSION TITLE
Can O3D3XX be used to write a program using ifm Vision Assistant to recognize different objects and measure the dimensions (length, width, height) of different parts of the object?

### DISCUSSION THREAD

**Discussion by mikufans-dumo** [ifm/ifm3d#467] [Q&A]
**_Project Background：_**
►Coffee Chairs, flower pots, and other handicrafts are suspended on the production line and sprayed by robotic arms. Before spraying, it is necessary to identify the type of handicraft through 3D sensors, and then measure the size of different parts (different types have different parts that need to be measured). If the size of a certain part does not meet the requirements, an alarm will be issued.
**_Schematic diagram of robotic arm spraying：_**
![Snipaste_2024-10-15_15-17-38](https://github.com/user-attachments/assets/414cd1e0-db82-49d3-bb49-cad2b4f83078)
**_Crafts to be sprayed：_**
![微信图片_20241015155811](https://github.com/user-attachments/assets/88a8326b-e824-4a23-a770-7bd2a69aa051)
![微信图片_20241015155807](https://github.com/user-attachments/assets/a0ed1dea-5bea-4a1e-b687-a1c578ac17f0)

**_For example：_**
►This is one of the chairs.We need to measure the dimensions above and below separately.
![Snipaste_2024-10-15_15-17-35](https://github.com/user-attachments/assets/9382db18-2f30-4ae7-a7b7-b3fc0f66e522)
![Snipaste_2024-10-15_15-17-23](https://github.com/user-attachments/assets/209dbe54-ed7e-4934-a442-a2f53cd0edb6)

_**Remarks:**_

- The handicrafts sprayed on the same production line are random and require cameras for type recognition. (Assuming that the minimum bounding rectangle size for each handicraft is unique)
- Different objects have different measurement requirements for different parts. (Not requiring high-precision measurements)
- The dimensions include: length, width, and height.
- Camera and object placement may have a certain rotation angle.

**Comment by lola-masson** [comment]
Hi @mikufans-dumo,

The O3D camera does not have any out-of-the-box algorithm that could handle this. 
It would certainly be possible to use the data for such use case, but you would need to do the implementation yourself.
